pkg/operator/ceph/csi/cluster_config.go

```diff
@@ -190,21 +190,17 @@ func CreateCsiConfigMap(namespace string, clientset kubernetes.Interface, ownerI
 // The locker l is typically a mutex and is used to prevent the config
 // map from being updated for multiple clusters simultaneously.
 func SaveClusterConfig(clientset kubernetes.Interface, clusterNamespace string, clusterInfo *cephclient.ClusterInfo, newCsiClusterConfigEntry *CsiClusterConfigEntry) error {
-	if !CSIEnabled() {
-		logger.Debug("skipping csi config update because csi is not enabled")
-		return nil
-	}
-
-	configMutex.Lock()
-	defer configMutex.Unlock()
-
 	// csi is deployed into the same namespace as the operator
 	csiNamespace := os.Getenv(k8sutil.PodNamespaceEnvVar)
 	if csiNamespace == "" {
-		return errors.Errorf("namespace value missing for %q", k8sutil.PodNamespaceEnvVar)
+		logger.Warningf("cannot save csi config due to missing env var %q", k8sutil.PodNamespaceEnvVar)
```

Comment:
Remind me how can we hit this? This code runs from the Operator so I don't see how we could enter this.

Reply:
Also, it doesn't seem related to the main goal of this patch, right?

Reply:
the same question is it possible to hit this one? this ENV will always be set in the rook operator pod isn't it?

Reply:
Exactly, it's always set in the operator pod at runtime. This change was just for unit tests. In unit tests when the mons are updated, the test was crashing because all the context wasn't properly initialized to allow updating this configmap. I decided to just return here for the unit tests, but we don't expect this condition in a real cluster.